src/server/src/modules/job.ts

```diff
@@ -0,0 +1,61 @@
+import { IJob, Job } from '@entities/job';
+import { getRepository } from 'typeorm';
+/**
+ * @description saves a new job in the database
+ * @param targetYears string[]
+ * @param hoursPerWeek number
+ * @param description string
+ * @param expirationDate Date
+ * @param startDate Date
+ * @param endDate Date
+ * @param type string[]
+ * @param title string
+ * @param status string
+ * @param minSalary number
+ * @param maxSalary number
+ * @param departmentId string
+ * @returns Promise
+ */
+export const createJob = (
+    targetYears: IJob['targetYears'],
+    hoursPerWeek: IJob['hoursPerWeek'],
+    description: IJob['description'],
+    expirationDate: IJob['expirationDate'],
+    startDate: IJob['startDate'],
+    endDate: IJob['endDate'],
+    type: IJob['type'],
+    title: IJob['title'],
+    status: IJob['status'],
+    minSalary: IJob['minSalary'],
+    maxSalary: IJob['maxSalary'],
+    departmentId: IJob['departmentId']
+) => {
+    if (expirationDate == null) {
+        // Set expiration date to 2 years after the current date.
+        expirationDate = new Date();
+        expirationDate.setFullYear(expirationDate.getFullYear() + 2);
+    }
+    if (endDate == null) {
+        // Set to expiration date
+        endDate = expirationDate;
```

Comment:
I'm not sure if we want to set the end date of the job to its expiration date. Mainly because the end date represents when the person who gets the job will stop working, and this doesn't have to be to years from now. This main lead to some confusion when jobs without end dates are posted. I think it will be better to leave as a NULL cell in the DB.

Reply:
Done.